arangodb-net-standard/CollectionApi/CollectionApiClient.cs

```diff
@@ -58,5 +58,16 @@ public async Task PutCollectionTruncateAsync(string collectionName)
             }
             await GetApiErrorException(response);
         }
+
+        public async Task<GetCollectionCountResponse> GetCollectionCountAsync(GetCollectionCountOptions options = null)
+        {
+            var response = await _transport.GetAsync(_collectionApiPath + "/" + options.CollectionName + "/count");
+            var stream = await response.Content.ReadAsStreamAsync();
+            if (response.IsSuccessStatusCode)
+            {                
+                return DeserializeJsonFromStream<GetCollectionCountResponse>(stream);
+            }
+            throw new ApiErrorException(DeserializeJsonFromStream<ApiErrorResponse>(stream));
```

Comment:
There's a convenience method now for constructing ApiErrorException from the response stream. You can use it like:
```
throw await GetApiErrorException(response);
```
This also makes it a good idea to only read the response stream inside the `if (response.IsSuccessStatusCode)` branch, to avoid two calls on `ReadAsStreamAsync`.  For example, the code for `DeleteCollectionAsync`:
```
using (var response = await _transport.DeleteAsync(_collectionApiPath + "/" + collectionName)) 
{
    if (response.IsSuccessStatusCode)
    {
        var stream = await response.Content.ReadAsStreamAsync();
        return DeserializeJsonFromStream<DeleteCollectionResponse>(stream, true, false);
    }
    throw await GetApiErrorException(response);
}
```

Reply:
I like this approach mainly for the fact that most of the error responses will always be serialized with the same set of parameters given to `DeserializeJsonFromStream`. So it's less error prone if it's encapsulated in a method. 👍